packages/graphql/src/schema/validation/custom-rules/valid-types/valid-list-in-node-type.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ASTVisitor, ObjectTypeDefinitionNode, TypeNode } from "graphql";
+import { Kind } from "graphql";
+import type { SDLValidationContext } from "graphql/validation/ValidationContext";
+import {
+    cypherDirective,
+    nodeDirective,
+    relationshipDirective,
+    relationshipPropertiesDirective,
+} from "../../../../graphql/directives";
+
+import { assertValid, createGraphQLError, DocumentValidationError } from "../utils/document-validation-error";
+import { getInnerTypeName } from "../utils/utils";
+
+/**
+ * Validates that list types used in type annotated with the node directive are supported by Neo4j
+ **/
+export function ValidListInNodeType(context: SDLValidationContext): ASTVisitor {
+    return {
+        ObjectTypeDefinition(objectTypeDefinitionNode: ObjectTypeDefinitionNode, _key, _parent) {
+            const { directives } = objectTypeDefinitionNode;
+
+            const nodeUsage = directives?.find((directive) => directive.name.value === nodeDirective.name);
+            const relationshipPropertiesUsage = directives?.find(
+                (directive) => directive.name.value === relationshipPropertiesDirective.name
+            );
+            if (!directives || (!nodeUsage && !relationshipPropertiesUsage)) {
+                return; // Skip non-node types or relationshipProperties types
+            }
```

Comment:
I think this is a bit more readable as 2 separate ifs. i.e. two cases we want to return